dataactcore/interfaces/function_bag.py

```diff
@@ -366,4 +366,25 @@ def get_submission_stats(submission_id):
         "total_obligations": float(base_query.scalar() or 0),
         "total_procurement_obligations": float(procurement.scalar() or 0),
         "total_assistance_obligations": float(fin_assist.scalar() or 0)
-    }
\ No newline at end of file
+    }
+
+
+def run_job_checks(job_id):
+    """ Checks that specified job has no unsatisfied prerequisites
+    Args:
+        job_id -- job_id of job to be run
+
+    Returns:
+        True if prerequisites are satisfied, False if not
+    """
+    sess = GlobalDB.db().session
+
+    # Get count of job's prerequisites that are not yet finished
+    incomplete_dependencies = sess.query(JobDependency). \
+        join("prerequisite_job"). \
+        filter(JobDependency.job_id == job_id, Job.job_status_id != JOB_STATUS_DICT['finished']). \
+        count()
+    if incomplete_dependencies:
+        return False
+    else:
+        return True
```

Comment:
Not sure if this is more legible, but it's conciser. What do you think?
```python
return not bool(incomplete_dependencies)
```
Maybe
```python
has_incomplete_dependencies = bool(incomplete_dependencies)
return not has_incomplete_dependencies
```

Unrelated, but agreed about not throwing an exception. It looks like everywhere we were using it, we were generating an exception _anyway_.